### PR TITLE
Adds support for PG15beta4

### PR DESCRIPTION
**Pull request by naisila**
Make sure we can properly build and all regression tests are passing.

Note: We have to manually make changes to ruleutils_15.c file.
Follow the relevant PG commit in ruleutils.c & make the exact same changes in ruleutils_15.c
In this case, the relevant PG commit is:
[96ef3237bf741c12390003e90a4d7115c0c854b7](https://github.com/postgres/postgres/commit/96ef3237bf741c12390003e90a4d7115c0c854b7)

### CODE REVIEW

File: src/backend/distributed/deparser/ruleutils_15.c [hunk 1]
```diff
@@ -430,12 +430,6 @@ static void get_coercion_expr(Node *arg, deparse_context *context,
 				  Node *parentNode);
 static void get_const_expr(Const *constval, deparse_context *context,
 			   int showtype);
-static void get_json_constructor(JsonConstructorExpr *ctor,
```
Comment (onderkalaci) [inline src/backend/distributed/deparser/ruleutils_15.c:433]: can you explain how you reverted the changes in this file? Mostly for future reference

**Review by hanefi — APPROVED** [review]
Good to merge after a rebase

**Review by onderkalaci — APPROVED** [review]
Locally tested check-multi, check-multi-1 and arbitrary configs.

Seems like src/test/regress/json_table_select_only.out and src/test/regress/json_table_select_only_0.out added to the repo accidentally on #6135

File: src/backend/distributed/deparser/ruleutils_15.c (hunk 1 above)
Comment (naisila) [reply src/backend/distributed/deparser/ruleutils_15.c:433]: Okay, putting that in the commit description and PR description as well

**Pull request merged**